server/config/config.go

```diff
@@ -117,44 +119,60 @@ func SetupFlags(flags *pflag.FlagSet) {
 	configOptions.registerFlags(flags)
 }
 
-func New(flags *pflag.FlagSet, logger logger) (*Config, error) {
-	vp := viper.New()
+type configOption func(*Config)
 
-	vp.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
-	vp.SetEnvPrefix("TRACETEST")
-	vp.AutomaticEnv()
+func WithLogger(l logger) configOption {
+	return func(c *Config) {
+		c.logger = l
+	}
+}
+
+func New(flags *pflag.FlagSet, confOpts ...configOption) (*Config, error) {
+	cfg := Config{
+		vp: viper.New(),
+	}
+
+	for _, coFn := range confOpts {
+		coFn(&cfg)
+	}
+
+	cfg.defaults()
```

Comment:
the new approach makes it easier to set defaults, which is great for things like test setups